digicampipe/utils/pulse_template.py

```diff
@@ -2,41 +2,123 @@
 import numpy as np
 from scipy.interpolate import interp1d
 
+from digicampipe.io.event_stream import calibration_event_stream
+from digicampipe.calib.baseline import fill_digicam_baseline, \
+    subtract_baseline, correct_wrong_baseline
+from digicampipe.calib.time import estimate_time_from_leading_edge
+from digicampipe.utils.hist2d import Histogram2dChunked
 
-class NormalizedPulseTemplate:
-    def __init__(self, amplitude, time):
-
-        dt = np.diff(time)
-
-        if not (dt == dt[0]).all():
-            raise ValueError('time argument should be of constant sampling')
 
+class NormalizedPulseTemplate:
+    def __init__(self, amplitude, time, amplitude_std=None):
         self.time = time
         self.amplitude = amplitude
+        if amplitude_std is not None:
+            assert len(amplitude_std) == len(amplitude)
```

Comment:
would be better to compare shapes since in principle this class works for 2 arrays as well
```suggestion
            assert amplitude_std.shape == amplitude.shape
```